### PR TITLE
Standardize on 18 character user ID

### PR DESCRIPTION
**Pull request by bbirman**
For #3211

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager+Internal.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+static NSString * const kOrgPrefix = @"00D";
+
+// Prefix of a user ID
+static NSString * const kUserPrefix = @"005";
+
+@interface SFDirectoryManager ()
+
+// Starting in SDK 8.2, 18 character IDs are used instead of 15 character IDs.
+// This renames 15 character user ID directories to 18 characters.
++ (void)upgradeUserDirectories;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager+Internal.h:40]: Add a `TODO:` here to remove this in `Mobile SDK 10.0`.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m [hunk 2]
```diff
@@ -235,4 +237,45 @@ - (void)migrateFiles {
     [sharedDefaults synchronize];
 }
 
++ (void)upgradeUserDirectories {
+    [SFDirectoryManager upgradeUserDirectory:NSLibraryDirectory];
+    [SFDirectoryManager upgradeUserDirectory:NSDocumentDirectory];
+}
+
++ (void)upgradeUserDirectory:(NSSearchPathDirectory)type {
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m:245]: Do we have files on iOS with orgid_userid  in the name (like we do on Android)?

Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m:245]: Could we add a test that exercise this method?

Comment (bhariharan) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m:245]: Oh, I should have mentioned this as well - we need upgrade steps for the Analytics data as well. The Analytics library takes a path from this library, so you'll need migration steps at both layers. I can talk to you offline about it.

Comment (bbirman) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m:245]: I only saw directories with the structure orgId/userId, though realized that I missed profile picture's userId (will add something for that). Is there anywhere else I should check for orgid_userid? Verified that the log files aren't dependent on userId and analytics's event_stores are covered by updating the userId directory in Documents. 
Screen Shot 2020-06-08 at 1 54 02 PM

Comment (bhariharan) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m:245]: Awesome! I can't think of anything else.

**Pull request merged**